src/devices/bus/coco/coco_sym12.cpp

```diff
@@ -0,0 +1,180 @@
+// license:BSD-3-Clause
+// copyright-holders:tim lidnner
+/***************************************************************************
+
+    coco_sym12.cpp
+
+
+	Code for emulating The Symphony 12
+
+	Made by Speech Systems, 1982.
+
+	This cartridge is a complex sound cartridge. It had 4 AY-3-8910 PSG
+	connected thru a PIA. It contained no ROM.
+
+***************************************************************************/
+
+#include "emu.h"
+#include "coco_sym12.h"
+#include "cococart.h"
+
+#include "speaker.h"
+#include "machine/6821pia.h"
+#include "sound/ay8910.h"
+
+//**************************************************************************
+//  SYMPHONY_TWELVE DEVICE CLASS
+//**************************************************************************
+
+namespace
+{
+	// ======================> coco_symphony_twelve_device
+
+	class coco_symphony_twelve_device :
+		public device_t,
+		public device_cococart_interface
+	{
+	public:
+		// construction/destruction
+		coco_symphony_twelve_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock)
+			: device_t(mconfig, COCO_SYM12, tag, owner, clock)
+			, device_cococart_interface(mconfig, *this)
+			, m_pia(*this, "s12_pia")
+			, m_ay8910(*this, "s12_ay8910-%u", 1)
+		{
+		}
+
+	protected:
+		// optional information overrides
+		virtual void device_add_mconfig(machine_config &config) override;
+
+		// device-level overrides
+		virtual void device_start() override
+		{
+			// install handlers
+ 			install_readwrite_handler( 0xFF60, 0xFF63, read8sm_delegate(*m_pia, FUNC(pia6821_device::read)), write8sm_delegate(*m_pia, FUNC(pia6821_device::write)));
```

Comment:
Please be consistent with hexadecimal case.  In your handler functions you’ve used lowercase, and most MAME code uses lowercase.  Please use lowercase here, too.